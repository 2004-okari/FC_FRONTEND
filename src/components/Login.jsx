import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { login } from '../redux/auth/authSlice';

const Login = () => {
  const dispatch = useDispatch();
  const { token, authenticate } = useSelector((store) => store.auth);
  const [username, setUsername] = useState('');

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(login(username));
  };

  return (
    <div className="flex items-center justify-center w-full px-3">
      <form className="flex flex-col gap-2 w-full items-center" onSubmit={handleLogin}>
        <input
          type="text"
          name="username"
          autoComplete="username"
          placeholder="username"
          className="rounded-lg border-2 w-[50%] p-1"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />

        <button
          type="submit"
          className="text-xl font-bold bg-orange-500 text-white rounded-lg py-1 w-[12%]"
        >
          Login
        </button>
      </form>
    </div>
  );
};

export default Login;
