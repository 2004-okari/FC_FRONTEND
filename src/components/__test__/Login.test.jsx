import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Login from '../Login';

const mockStore = configureStore([]);

describe('Login Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      auth: {
        token: null,
      },
    });

    store.dispatch = jest.fn();

    component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('renders correctly', () => {
    expect(component.container).toMatchSnapshot();
  });
});
