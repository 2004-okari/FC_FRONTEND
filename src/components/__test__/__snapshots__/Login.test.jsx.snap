// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<div>
  <div
    class="flex items-center justify-center w-full px-3 h-screen"
  >
    <form
      class="flex flex-col gap-8 w-full items-center"
    >
      <input
        autocomplete="username"
        class="rounded-lg border-2 w-3/4 md:w-[50%] p-1"
        name="username"
        placeholder="Enter your username to login"
        type="text"
        value=""
      />
      <button
        class="text-xl bg-green-600 hover:bg-green-700 text-white duration-300 rounded-lg py-1 w-24"
        type="submit"
      >
        Login
      </button>
      <p>
        <span
          class="text-gray-400"
        >
          New to MotorBook? 
        </span>
        <a
          class="hover:underline"
          href="/signup"
        >
          Sign up now
        </a>
      </p>
    </form>
  </div>
</div>
`;
